### ISSUE TITLE
reconnect during ibd does not resume sync where we left off

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#5145]
Observed on 115da942

>2023-07-11 20:57:49,295UTC INFO [BitcoinSServerMain] version=1.9.7-208-115da942-20230711-1554-SNAPSHOT jdkVersion=21-ea

```
2023-07-12 01:50:29,185UTC INFO [DLCWallet$DLCWalletImpl] Finished processing of block=0000000000000000000fc3cd300331260715069ebce905aef4c2f65d57a66337. It took 5631ms
2023-07-12 01:50:29,189UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000010929ee359dd323a4343e72f5eec8d4575461a4025c21c heightOpt=Some(697470)
2023-07-12 01:50:34,866UTC INFO [DLCWallet$DLCWalletImpl] Finished processing of block=00000000000000000010929ee359dd323a4343e72f5eec8d4575461a4025c21c. It took 5680ms
2023-07-12 01:50:34,870UTC INFO [DLCWallet$DLCWalletImpl] Processing block=0000000000000000000885f967839bb0ab383ac80c3859ea6471fe503b387456 heightOpt=Some(697566)
2023-07-12 01:50:53,423UTC INFO [DLCWallet$DLCWalletImpl] Finished processing of block=0000000000000000000885f967839bb0ab383ac80c3859ea6471fe503b387456. It took 18556ms
2023-07-12 01:50:53,427UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000007d7b0e314c8a5b76dc8bd02fbd3923afeee475e83b4eb heightOpt=Some(697605)
2023-07-12 01:51:06,554UTC INFO [DLCWallet$DLCWalletImpl] Finished processing of block=00000000000000000007d7b0e314c8a5b76dc8bd02fbd3923afeee475e83b4eb. It took 13130ms
2023-07-12 01:53:10,223UTC INFO [PeerFinder] Attempting to reconnect peer=Peer(neutrino.suredbits.com:8333)
2023-07-12 01:53:10,756UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(neutrino.suredbits.com:8333)
2023-07-12 01:53:10,759UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:43990
2023-07-12 01:53:12,147UTC INFO [Socks5Connection$] Tor connection request succeeded. target=neutrino.suredbits.com/<unresolved>:8333 connectedAddress=/0.0.0.0:0
2023-07-12 01:53:12,902UTC INFO [PeerManager] Connected to peer Peer(neutrino.suredbits.com:8333) with filters. Connected peer count 1
```

This is where things go wrong, we don't resume syncing from our prior spot after reconnect is done

```
2023-07-12 02:05:30,480UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000002c216d341c573665bb46b22767e5a1f8b49dc1afcafe3
2023-07-12 02:05:30,493UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000002c216d341c573665bb46b22767e5a1f8b49dc1afcafe3, skipping processing in wallet
2023-07-12 02:05:30,493UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000002c216d341c573665bb46b22767e5a1f8b49dc1afcafe3 heightOpt=None
2023-07-12 02:12:38,712UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000014f8cacd99a15b5488282b33a913199a320c56b417ca9
2023-07-12 02:12:38,720UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000014f8cacd99a15b5488282b33a913199a320c56b417ca9, skipping processing in wallet
2023-07-12 02:12:38,720UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000014f8cacd99a15b5488282b33a913199a320c56b417ca9 heightOpt=None
2023-07-12 02:26:25,946UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000025adae7bc875e5e928cf18d173732089722702c2448b1
2023-07-12 02:26:25,955UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000025adae7bc875e5e928cf18d173732089722702c2448b1, skipping processing in wallet
2023-07-12 02:26:25,955UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000025adae7bc875e5e928cf18d173732089722702c2448b1 heightOpt=None
2023-07-12 02:28:23,315UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000003a3bbc3dac0c578948918c796426cda4c1958fc8454d3
2023-07-12 02:28:23,323UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000003a3bbc3dac0c578948918c796426cda4c1958fc8454d3, skipping processing in wallet
2023-07-12 02:28:23,323UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000003a3bbc3dac0c578948918c796426cda4c1958fc8454d3 heightOpt=None
2023-07-12 02:29:56,837UTC INFO [PeerFinder] Querying p2p network for peers...
2023-07-12 02:29:56,857UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(86.150.55.103:8333)
2023-07-12 02:29:56,859UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:60388
2023-07-12 02:29:56,867UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(91.21.178.106:8333)
2023-07-12 02:29:56,869UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:60402
2023-07-12 02:29:56,875UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(37.71.249.42:8333)
2023-07-12 02:29:56,876UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:60404
2023-07-12 02:29:56,886UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(107.115.32.162:8333)
2023-07-12 02:29:56,888UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:60418
2023-07-12 02:29:56,897UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(181.41.202.99:8333)
2023-07-12 02:29:56,899UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:60420
2023-07-12 02:29:56,905UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(93.189.81.122:8333)
2023-07-12 02:29:56,907UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:60422
2023-07-12 02:29:56,912UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(185.174.108.19:8333)
2023-07-12 02:29:56,913UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:60426
2023-07-12 02:29:56,918UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer([2401:7000:ca2e:4400:dcba:285:b024:4bc2]:8333)
2023-07-12 02:29:56,919UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:60430
2023-07-12 02:30:06,877UTC INFO [PeerMessageSender] Disconnecting peer=Peer(86.150.55.103:8333)
2023-07-12 02:30:06,887UTC INFO [PeerMessageSender] Disconnecting peer=Peer(91.21.178.106:8333)
2023-07-12 02:30:06,896UTC INFO [PeerMessageSender] Disconnecting peer=Peer(37.71.249.42:8333)
2023-07-12 02:30:06,906UTC INFO [PeerMessageSender] Disconnecting peer=Peer(107.115.32.162:8333)
2023-07-12 02:30:06,916UTC INFO [PeerMessageSender] Disconnecting peer=Peer(181.41.202.99:8333)
2023-07-12 02:30:06,926UTC INFO [PeerMessageSender] Disconnecting peer=Peer(93.189.81.122:8333)
2023-07-12 02:30:06,927UTC INFO [PeerMessageSender] Disconnecting peer=Peer(185.174.108.19:8333)
2023-07-12 02:30:06,936UTC INFO [PeerMessageSender] Disconnecting peer=Peer([2401:7000:ca2e:4400:dcba:285:b024:4bc2]:8333)
2023-07-12 02:30:18,845UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000003f5303e4e1f069d5875557fe99ed9f61ce8b8998d2006
2023-07-12 02:30:18,849UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000003f5303e4e1f069d5875557fe99ed9f61ce8b8998d2006, skipping processing in wallet
2023-07-12 02:30:18,849UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000003f5303e4e1f069d5875557fe99ed9f61ce8b8998d2006 heightOpt=None
2023-07-12 02:30:33,348UTC INFO [DataMessageHandler] Received block message with hash 0000000000000000000517abcd1a2d35d313028d4252d6a97633ed1c42742370
2023-07-12 02:30:33,352UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=0000000000000000000517abcd1a2d35d313028d4252d6a97633ed1c42742370, skipping processing in wallet
2023-07-12 02:30:33,353UTC INFO [DLCWallet$DLCWalletImpl] Processing block=0000000000000000000517abcd1a2d35d313028d4252d6a97633ed1c42742370 heightOpt=None
2023-07-12 02:43:21,505UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000002fd701129f8eca5545daa28c93ac072cb7bb5eb1e9382
2023-07-12 02:43:21,513UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000002fd701129f8eca5545daa28c93ac072cb7bb5eb1e9382, skipping processing in wallet
2023-07-12 02:43:21,513UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000002fd701129f8eca5545daa28c93ac072cb7bb5eb1e9382 heightOpt=None
2023-07-12 02:45:36,620UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000004916cd6c672b818efb5646aec524440dbef9d107d0433
2023-07-12 02:45:36,625UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000004916cd6c672b818efb5646aec524440dbef9d107d0433, skipping processing in wallet
2023-07-12 02:45:36,625UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000004916cd6c672b818efb5646aec524440dbef9d107d0433 heightOpt=None
2023-07-12 02:55:34,241UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000043011bc05524f3edf8627e069a9f5b5a828475f1957d8
2023-07-12 02:55:34,248UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000043011bc05524f3edf8627e069a9f5b5a828475f1957d8, skipping processing in wallet
2023-07-12 02:55:34,248UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000043011bc05524f3edf8627e069a9f5b5a828475f1957d8 heightOpt=None
2023-07-12 03:07:23,501UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000004c3dd1955ae57680c9f9a6bc2ea4140f7461604d3cb51
2023-07-12 03:07:23,509UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000004c3dd1955ae57680c9f9a6bc2ea4140f7461604d3cb51, skipping processing in wallet
2023-07-12 03:07:23,509UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000004c3dd1955ae57680c9f9a6bc2ea4140f7461604d3cb51 heightOpt=None
2023-07-12 03:15:40,414UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000005180b03ada74e79274eb56dbd4147fb05b0ed4d9a0189
2023-07-12 03:15:40,421UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000005180b03ada74e79274eb56dbd4147fb05b0ed4d9a0189, skipping processing in wallet
2023-07-12 03:15:40,421UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000005180b03ada74e79274eb56dbd4147fb05b0ed4d9a0189 heightOpt=None
2023-07-12 03:29:36,802UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000021d7926df8b1a267f37effba98f2754b4fa2587e72ad5
2023-07-12 03:29:36,808UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000021d7926df8b1a267f37effba98f2754b4fa2587e72ad5, skipping processing in wallet
2023-07-12 03:29:36,808UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000021d7926df8b1a267f37effba98f2754b4fa2587e72ad5 heightOpt=None
2023-07-12 03:29:56,936UTC INFO [PeerFinder] Querying p2p network for peers...
2023-07-12 03:29:56,959UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer([2600:1f18:66fc:d701:d68d:ea70:77a0:4072]:8333)
2023-07-12 03:29:56,961UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:33694
2023-07-12 03:29:56,968UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(150.117.54.192:8333)
2023-07-12 03:29:56,970UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:33702
2023-07-12 03:29:56,975UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(170.75.161.146:39388)
2023-07-12 03:29:56,976UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:33712
2023-07-12 03:29:56,981UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(91.3.180.188:8333)
2023-07-12 03:29:56,983UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:33716
2023-07-12 03:29:56,988UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(73.17.132.227:8333)
2023-07-12 03:29:56,990UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:33726
2023-07-12 03:29:56,994UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer([2a01:4f9:3a:2dd2:0:0:0:2]:8333)
2023-07-12 03:29:56,996UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:33734
2023-07-12 03:29:57,000UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(133.18.228.108:8333)
2023-07-12 03:29:57,002UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:33738
2023-07-12 03:29:57,006UTC INFO [PeerMessageSender] Attempting to connect to peer=Peer(71.121.205.52:8333)
2023-07-12 03:29:57,008UTC INFO [PeerMessageSender] Connected to remote=localhost/127.0.0.1:1635  local=/127.0.0.1:33748
2023-07-12 03:29:57,355UTC INFO [Socks5Connection$] Tor connection request succeeded. target=91.3.180.188/<unresolved>:8333 connectedAddress=/0.0.0.0:0
2023-07-12 03:29:57,355UTC INFO [Socks5Connection$] Tor connection request succeeded. target=[2a01:4f9:3a:2dd2:0:0:0:2]/<unresolved>:8333 connectedAddress=/0.0.0.0:0
2023-07-12 03:29:57,368UTC INFO [Socks5Connection$] Tor connection request succeeded. target=[2600:1f18:66fc:d701:d68d:ea70:77a0:4072]/<unresolved>:8333 connectedAddress=/0.0.0.0:0
2023-07-12 03:29:57,545UTC INFO [Socks5Connection$] Tor connection request succeeded. target=71.121.205.52/<unresolved>:8333 connectedAddress=/0.0.0.0:0
2023-07-12 03:29:57,639UTC INFO [Socks5Connection$] Tor connection request succeeded. target=133.18.228.108/<unresolved>:8333 connectedAddress=/0.0.0.0:0
2023-07-12 03:30:07,658UTC INFO [PeerMessageSender] Disconnecting peer=Peer([2a01:4f9:3a:2dd2:0:0:0:2]:8333)

```

**Comment by Christewart** [comment]
Related to #5154

**Comment by Christewart** [comment]
fixed in #5154
